src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: Function,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-}
+    const biggestSide = Math.max(this.a, this.b, this.c);
+    const sumOfSmallerSides = [this.a, this.b, this.c]
+      .reduce((sum, curr) => sum + curr) - biggestSide;
+
+    if (biggestSide >= sumOfSmallerSides) {
+      throw new Error('The longest side of the triangle must be less'
+      + 'than the sum of the other two');
+    }
+
+    if (this.a < 0 || this.b < 0 || this.c < 0
+      || biggestSide >= sumOfSmallerSides) {
+      throw new Error('Sides of the triangle must be bigger than 0');
+    }
+  }
+
+  shape: Shape;
 
-export class Circle {
+  getArea(): number {
+    const avergeSide = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(Math.sqrt(
+      avergeSide * (avergeSide - this.a)
+      * (avergeSide - this.b) * (avergeSide - this.c),
+    ) * 100) / 100;
```

Comment:
Consider using a function for rounding since you apply it three times.

Reply:
agree